packages/report-server/src/reportBuilder/transform/parser/TransformParser.ts

```diff
@@ -65,14 +77,22 @@ export class TransformParser extends ExpressionParser {
       });
       this.set('where', this.whereFunction); // no '@' prefix for where
     }
+
+    this.context = context;
   }
 
-  public evaluate(expression: string) {
-    if (TransformParser.isExpression(expression)) {
-      return super.evaluate(expression.substring(1));
-    }
+  public isExpression(input: unknown) {
+    return typeof input === 'string' && input.startsWith(this.EXPRESSION_PREFIX);
+  }
 
-    return expression;
+  protected readExpression(input: unknown) {
+    return this.isExpression(input)
+      ? (input as string).replace(new RegExp(`^${this.EXPRESSION_PREFIX}`), '')
+      : input;
+  }
```

Comment:
Eventually I followed your suggestion and moved most of the logic in `TransformParser` 🙂 Not doing so would result in some interesting questions which we probably cannot answer consistently at this stage:
  * What happens if something is not an expression? Do we throw an error or just use the input?
   * Similarly: is the prefix optional or not? And if it is optional, what's the point in using it?

The compromise I thought of is adding a `readExpression` method in the base class, which can be overridden to support custom expression formats. The caveat here is that future devs must remember to call this function when an expression needs to be read in `ExpressionParser`. It's not ideal, but extending all present (and future) `ExpressionParser` methods in `TransformParser` so that our custom expression parsing is used seems even worse!

Another advantage of this pattern is that the custom format specified in sub classes is not restricted to a prefix, but could be anything.

Reply:
Regarding semantics, I still believe that anything that can be evaluated is an _expression_. Following that definition, we should treat all expressions equally:
* for the literal number `2`: use `=2`
* for the literal string `MyString`: use `='MyString'`, or `=MyString` if our parser can easily distinguish between variables, functions and string literals

**But** I guess we had a valid reason for not doing so - I remember that in the past we used expressions like the following in the left sides of params, which didn't read fluently:
`"'name'":  "orgUnit"`

Reply:
I believe that left-side evaluation is what what makes a difference in `report-server` in relation to other languages we are (I am?) used into, and may be the reason - and justification! of more custom expression parsing.

Eg in Javascript:
```js
const valid = { result: 2 + 3 };
const invalid = { 2 + 3: 'result' };
```